### PR TITLE
Allow * as a device name in grm

### PR DESCRIPTION
**Pull request by orenc17**
### Description

Allowing * as device name in grm mode, this will scan the test_spec.json and add all the platform required as DUMMY platforms

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [X] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### CODE REVIEW

**Comment by orenc17** [comment]
@bridadan @theotherjimmy tested on my machine

**Comment by jupe** [comment]
is it idea to support parallel execution with grm?

File: packages/mbed-greentea/mbed_greentea/mbed_greentea_cli.py [hunk 1]
```diff
@@ -686,8 +686,14 @@ def get_parallel_value(value):
             gt_logger.gt_log_warn("entering global resource manager mbed-ls dummy mode!")
             grm_platform_name, grm_module_name, grm_ip_name, grm_port_name = grm_values
             mbeds_list = []
-            for _ in range(parallel_test_exec):
-                mbeds_list.append(mbeds.get_dummy_platform(grm_platform_name))
+            if grm_platform_name is '*':
```
Comment (bridadan) [inline packages/mbed-greentea/mbed_greentea/mbed_greentea_cli.py]: I think literal comparisons are frowned upon in Python (my linter always yells at me for doing stuff like this).
```suggestion
            if grm_platform_name == '*':
```

**Comment by bridadan** [comment]
@jupe this is for PSA targets that currently have differently named targets in Mbed OS than how our tools detect them: `TARGET_NS`, `TARGET_S`, `TARGET`, etc

**Comment by orenc17** [comment]
@jupe  it could be

**Pull request merged**